diff -p -up vdr-1.6.0/channels.c.orig vdr-1.6.0/channels.c
--- vdr-1.6.0/channels.c.orig	2008-03-05 18:42:50.000000000 +0200
+++ vdr-1.6.0/channels.c	2009-03-21 00:18:07.000000000 +0200
@@ -574,7 +574,7 @@ void cChannel::SetLinkChannels(cLinkChan
      }
   else
      q += sprintf(q, " none");
-  dsyslog(buffer);
+  dsyslog("%s", buffer);
 }
 
 void cChannel::SetRefChannel(cChannel *RefChannel)
diff -p -up vdr-1.6.0/PLUGINS/src/sky/sky.c.orig vdr-1.6.0/PLUGINS/src/sky/sky.c
--- vdr-1.6.0/PLUGINS/src/sky/sky.c.orig	2008-03-22 12:19:32.000000000 +0200
+++ vdr-1.6.0/PLUGINS/src/sky/sky.c	2009-03-21 00:18:07.000000000 +0200
@@ -110,7 +110,7 @@ void cDigiboxDevice::LircSend(const char
 {
   char buf[100];
   snprintf(buf, sizeof(buf), "SEND_ONCE SKY %s\n", s);
-  dsyslog(buf);//XXX
+  dsyslog("%s", buf);//XXX
   if (write(fd_lirc, buf, strlen(buf)) < 0)
      LOG_ERROR;//XXX _STR
   cCondWait::SleepMs(200);
diff -p -up vdr-1.6.0/receiver.c.orig vdr-1.6.0/receiver.c
--- vdr-1.6.0/receiver.c.orig	2007-08-12 14:52:59.000000000 +0300
+++ vdr-1.6.0/receiver.c	2009-03-21 00:18:07.000000000 +0200
@@ -40,7 +40,7 @@ cReceiver::~cReceiver()
 {
   if (device) {
      const char *msg = "ERROR: cReceiver has not been detached yet! This is a design fault and VDR will segfault now!";
-     esyslog(msg);
+     esyslog("%s", msg);
      fprintf(stderr, "%s\n", msg);
      *(char *)0 = 0; // cause a segfault
      }
