diff -Nru vdr-1.6.0-vanilla/svdrp.c vdr-1.6.0-hitk/svdrp.c
--- vdr-1.6.0-vanilla/svdrp.c	2008-02-17 15:36:01.000000000 +0200
+++ vdr-1.6.0-hitk/svdrp.c	2009-03-29 18:29:20.000000000 +0300
@@ -893,7 +893,10 @@
   if (*Option) {
      eKeys k = cKey::FromString(Option);
      if (k != kNone) {
-        cRemote::Put(k);
+        if (KeyMacros.Get(k))
+           cRemote::PutMacro(k);
+        else
+           cRemote::Put(k);
         Reply(250, "Key \"%s\" accepted", Option);
         }
      else
