Index: vdr-1.6.0-fonta/menu.c
===================================================================
--- vdr-1.6.0-fonta/menu.c
+++ vdr-1.6.0-fonta/menu.c	2008-04-25 16:35:17.000000000 +0300
@@ -2227,9 +2227,12 @@
   cFont::GetAvailableFontNames(&fontOsdNames);
   cFont::GetAvailableFontNames(&fontSmlNames);
   cFont::GetAvailableFontNames(&fontFixNames, true);
-  fontOsdNames.Insert(strdup(DefaultFontOsd));
-  fontSmlNames.Insert(strdup(DefaultFontSml));
-  fontFixNames.Insert(strdup(DefaultFontFix));
+  fontOsdNames.Insert(strdup("Sans Serif"));
+  fontOsdNames.Insert(strdup("Sans Serif:Bold"));
+  fontSmlNames.Insert(strdup("Sans Serif"));
+  fontSmlNames.Insert(strdup("Sans Serif:Bold"));
+  fontFixNames.Insert(strdup("Monospace"));
+  fontFixNames.Insert(strdup("Monospace:Bold"));
   fontOsdIndex = max(0, fontOsdNames.Find(Setup.FontOsd));
   fontSmlIndex = max(0, fontSmlNames.Find(Setup.FontSml));
   fontFixIndex = max(0, fontFixNames.Find(Setup.FontFix));
