--- vdr-1.6.0/tools.h.version	2008-02-17 14:41:27.000000000 +0100
+++ vdr-1.6.0/tools.h	2009-02-06 13:16:43.000000000 +0100
@@ -205,6 +205,7 @@
 bool SpinUpDisk(const char *FileName);
 void TouchFile(const char *FileName);
 time_t LastModifiedTime(const char *FileName);
+const char *VdrVersion(void);
 cString WeekDayName(int WeekDay);
 cString WeekDayName(time_t t);
 cString WeekDayNameFull(int WeekDay);
--- vdr-1.6.0/tools.c~	2008-03-05 18:23:47.000000000 +0100
+++ vdr-1.6.0/tools.c	2009-02-06 13:23:26.000000000 +0100
@@ -27,6 +27,7 @@
 #include <utime.h>
 #include "i18n.h"
 #include "thread.h"
+#include "config.h"
 
 int SysLogLevel = 3;
 
@@ -540,6 +541,11 @@
   return 0;
 }
 
+const char *VdrVersion(void)
+{
+	return VDRVERSION;
+}	 
+
 // --- cTimeMs ---------------------------------------------------------------
 
 cTimeMs::cTimeMs(int Ms)
