Index: vdr-1.6.0/h264parser.c
===================================================================
--- vdr-1.6.0/h264parser.c
+++ vdr-1.6.0/h264parser.c	2009-07-28 21:01:11.182093897 +0300
@@ -445,7 +445,7 @@
               }
           catch (cException *e) {
               if (!syncing) // suppress typical error messages while syncing
-                 esyslog("%s", e->Message());
+                 esyslog("%s", (const char*) e->Message());
               delete e;
               }
           }
